packages/datadog-plugin-langchain/src/handlers/embedding.js

```diff
@@ -0,0 +1,60 @@
+'use strict'
+
+const LangChainHandler = require('./default')
+
+class LangChainEmbeddingHandler extends LangChainHandler {
+  getSpanStartTags (ctx) {
+    const tags = {}
+
+    const inputTexts = ctx.args?.[0]
+
+    if (typeof inputTexts === 'string') {
+      // embed query
+      if (this.isPromptCompletionSampled()) {
+        tags['langchain.request.inputs.0.text'] = this.truncate(inputTexts)
+      }
+      tags['langchain.request.input_counts'] = 1
+    } else {
+      // embed documents
+      if (this.isPromptCompletionSampled()) {
+        for (const idx in inputTexts) {
+          const inputText = inputTexts[idx]
+          tags[`langchain.request.inputs.${idx}.text`] = this.truncate(inputText)
+        }
+      }
+      tags['langchain.request.input_counts'] = inputTexts.length
+    }
+
+    return tags
+  }
+
+  getSpanEndTags (ctx) {
+    const tags = {}
+
+    const { result } = ctx
+    if (!Array.isArray(result)) return
+
+    if (Array.isArray(result[0])) {
+      for (const idx in result) {
+        const output = result[idx]
+        tags[`langchain.response.outputs.${idx}.embedding_length`] = output.length
+      }
+    } else {
+      tags['langchain.response.outputs.embedding_length'] = result.length
+    }
```

Comment:
This is fine as is, but I wonder if we should just tag one embedding length tag instead of one per entry since every model will output the same dimension embeddings for every input (since dimensions are a model-specific param AFAIK).

Reply:
ah yes good point. I think I lifted this right from the [Python integration](https://github.com/DataDog/dd-trace-py/blob/ac94050bb86b337dddb85f6fc2fa29d08c809b15/ddtrace/contrib/internal/langchain/patch.py#L630-L634), so as long we're ok deviating here, I think it probably does make more sense to just have one tag

Reply:
Haha yup I noticed we do the same in the Python integration, but wanted to make sure we don't repeat the same mistakes when possible (it'll be harder to remove this from the Python integration but I'd rather we not have this unnecessary tagging to begin with)

Reply:
yep makes sense. i added it in [this commit](https://github.com/DataDog/dd-trace-js/pull/4860/commits/69d88fb9b03ff3bc3720a76fd051fc150cedad92#diff-ea314d2f0583dc2d704f638a312b91f0d82ba1f532123b6b0fa8c1831637e93f)